lisa/tests/kernel/scheduler/load_tracking.py

```
@@ -733,3 +735,218 @@ def test_load_avg_behaviour(self, error_margin_pct=5, allowed_error_pct=5)\
+
+        :param allowed_error_pct: How much the trace averages can stray from the
+          expected values
+        :type allowed_error_pct: float
+        """
+        expected_cpu_util = self.get_expected_cpu_util()
+        trace_cpu_util = self.get_trace_cpu_util()
+
+        passed = True
+
+        expected_metrics = {}
+        trace_metrics = {}
+        deltas = {}
+
+        for cpu in self.cpus:
+            cpu_str = "cpu{}".format(cpu)
+
+            expected_metrics[cpu_str] = TestMetric({})
+            trace_metrics[cpu_str] = TestMetric({})
+            deltas[cpu_str] = TestMetric({})
+
+            for phase in range(self.nr_phases):
+                if not self.is_almost_equal(
+                        trace_cpu_util[cpu][phase],
+                        expected_cpu_util[cpu][phase],
+                        allowed_error_pct):
+                    passed = False
+
+                # Just some verbose metric collection...
+                phase_str = "phase{}".format(phase)
+
+                expected_metrics[cpu_str].data[phase_str] = TestMetric(
+                    expected_cpu_util[cpu][phase])
+                trace_metrics[cpu_str].data[phase_str] = TestMetric(
+                    trace_cpu_util[cpu][phase])
+                deltas[cpu_str].data[phase_str] = TestMetric(
+                    -100 * (expected_cpu_util[cpu][phase] - trace_cpu_util[cpu][phase]) /
+                    expected_cpu_util[cpu][phase], "%")
+
+        res = ResultBundle.from_bool(passed)
+        res.add_metric("Expected utilization", expected_metrics)
+        res.add_metric("Trace utilization", trace_metrics)
+        res.add_metric("Utilization deltas", deltas)
+
+        return res
+
+class OneTaskCPUMigration(CPUMigrationBase):
+    """
+    Some tasks on two big CPUs, one of them migrates in its second phase.
+    """
+
+    @classmethod
+    def get_migration_cpus(cls, te):
+        cpu_capacities = te.plat_info["cpu-capacities"]
```

Wondering if we should add in some check/errors/failure in case this info is not available.
Not sure if we have any other pre-check before executing a test that all the required data are there... but I would expect in that case we expose somehow what are the requirements for a test... and we don't do that here AFAIKS.

---

That key should always be there since it is computed from the target. If it is not, it will fail, saying that cpu-capacities cannot be found anyway. If we can provide more information by failing early, we could add more pre-tests, but otherwise I don't really see the point (it will never be silent in any case).

---

Mmm yeah, I did add this hook: https://github.com/ARM-software/lisa/blob/next/lisa/tests/kernel/test_bundle.py#L264, but I haven't been a perfect citizen in that I don't think I've used it everywhere I should've...